src/platform/macos/view.rs

```diff
@@ -351,8 +351,27 @@ extern fn key_down(this: &Object, _sel: Sel, event: id) {
         let state = &mut *(state_ptr as *mut ViewState);
         let window_id = WindowId(get_window_id(state.window));
 
+        state.raw_characters = {
```

Comment:
It would be really nice to have the code for reading the characters to a `String` put into a function, since it's repeated now.